milli/src/update/available_documents_ids.rs

```diff
@@ -8,11 +8,16 @@ pub struct AvailableDocumentsIds {
 }
 
 impl AvailableDocumentsIds {
-    pub fn from_documents_ids(docids: &RoaringBitmap) -> AvailableDocumentsIds {
-        match docids.max() {
+    pub fn from_documents_ids(
+        docids: &RoaringBitmap,
+        soft_deleted_docids: &RoaringBitmap,
+    ) -> AvailableDocumentsIds {
+        let used_docids = docids | soft_deleted_docids;
+
+        match used_docids.max() {
             Some(last_id) => {
                 let mut available = RoaringBitmap::from_iter(0..last_id);
-                available -= docids;
+                available -= used_docids;
```

Comment:
Could you do that in one line, please? Put the OR on the `used_docids` line?

Reply:
Not sure I understood. Do you want something like that 🤔?
```rust
available -= docids | soft_deleted_docids;
```

If yes, did you miss the fact that I also use the `used_docids` in the match?